dali/ft/filecopy.cpp

```diff
@@ -1943,9 +1998,11 @@ void FileSprayer::cloneHeaderFooter(unsigned idx, bool isHeader)
     PartitionPoint & next = * new PartitionPoint;
     //NB: headerSize include the size of the xmlHeader; size includes neither header or footers.
     if (isHeader)
+        // Set offset to the XML header
         next.inputOffset = curSrc.headerSize - curSrc.xmlHeaderLength;
     else
-        next.inputOffset = curSrc.headerSize + curSrc.size;
+        //Set offset to the XML footer
+        next.inputOffset = curSrc.size + curSrc.xmlHeaderLength;
```

Comment:
Also did you test this with xml files that start with a BOM?  I think curSrc.headerSize includes the BOM but don't think curSrc.xmlHeaderLength does.